compiler/plugins/input/StableHLO/Conversion/StableHLOToLinalgExt.cpp

```diff
@@ -427,22 +428,60 @@ struct FftOpConversion final : OpConversionPattern<mlir::stablehlo::FftOp> {
 struct ReverseOpConversion final
     : OpConversionPattern<mlir::stablehlo::ReverseOp> {
   using OpConversionPattern::OpConversionPattern;
-
   LogicalResult
   matchAndRewrite(mlir::stablehlo::ReverseOp op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
-    auto ty = dyn_cast<RankedTensorType>(adaptor.getOperands()[0].getType());
+    const auto ty =
+        dyn_cast<RankedTensorType>(adaptor.getOperands()[0].getType());
     if (!ty)
       return failure();
 
-    Location loc = op.getLoc();
-    SmallVector<OpFoldResult> mixedSizes =
-        tensor::getMixedSizes(rewriter, loc, adaptor.getOperands()[0]);
-    Value emptyTensor =
-        rewriter.create<tensor::EmptyOp>(loc, mixedSizes, ty.getElementType());
-    rewriter.replaceOpWithNewOp<IREE::LinalgExt::ReverseOp>(
-        op, typeConverter->convertType(op.getType()), adaptor.getOperands(),
-        emptyTensor, rewriter.getI64TensorAttr(op.getDimensions()));
+    Value input = op.getOperand();
+    const auto inputTy = cast<ShapedType>(input.getType());
+    const auto resultTy = cast<ShapedType>(op.getType());
+    const auto dims = op.getDimensions();
+    const Location loc = op.getLoc();
+    const auto inputTyRank = inputTy.getRank();
```

Comment:
I like `const`, but we usually don't add const to `Type`, `Location`, and rank. Can we drop them for consistency in the codebase?

```suggestion
    auto inputTy = cast<ShapedType>(input.getType());
    auto resultTy = cast<ShapedType>(op.getType());
    const auto dims = op.getDimensions(); // Can this be ArrayRef?
    Location loc = op.getLoc();
    int64_t inputTyRank = inputTy.getRank();
```

Reply:
updated